kernel/test/KernelRegistry.test.js

```diff
@@ -0,0 +1,53 @@
+import assertRevert from './helpers/assertRevert';
+
+const KernelInstance = artifacts.require('KernelInstance');
+const KernelRegistry = artifacts.require('KernelRegistry');
+
+require('chai')
+  .use(require('chai-as-promised'))
+  .should();
+
+contract('KernelRegistry', ([owner, other]) => {
+  const name = "Test";
+  const version_0 = "0.0";
+  const version_1 = "0.1";
+
+  beforeEach(async function () {
+    this.kernelInstance = await KernelInstance.new(name, version_0, 0);
+    await this.kernelInstance.freeze();
+    this.kernelInstance2 = await KernelInstance.new(name, version_1, this.kernelInstance.address);
+    this.kernelRegistry = await KernelRegistry.new();
+  });
+
+  it('should add single instance', async function () {
+    await this.kernelRegistry.addInstance(this.kernelInstance.address).should.be.fulfilled;
```

Comment:
shouldn't we check that `KernelRegistry#getInstance` works as expected after adding a new instance?

Reply:
mm not sure what you mean, isn't this tested in `'should correctly recover instances'`? Or are you saying after adding _two_ instances?